File: sessions/src/main/java/com/planorama/controller/response/SessionResponse.java
package com.planorama.controller.response;

import com.planorama.domain.Captions;
import com.planorama.domain.User;

import java.time.LocalDateTime;

public record SessionResponse(String id,
                              String scheduleId,
                              String ownerId,
                              String title,
                              String startTime,
                              String endTime,
                              String description,
                              Captions caption,
                              User patient) {
}

File: planorama-frontend/src/app/components/modal/domain/session_response.ts
import { Captions } from './captions';

export class SessionResponse {
  public id!: string;
  public scheduleId!: string;
  public ownerId!: string;
  public title!: string;
  public startTime!: Date;
  public endTime!: Date;
  public description!: string;
  public caption!: Captions;
  public color!: Color;
}

export class Color {
  public primary: string;
  public secondary: string;

  constructor(primary: string, secondary: string) {
    this.primary = primary;
    this.secondary = secondary;
  }
}
